Sources/DistributedActors/InvocationBehavior.swift

```diff
@@ -32,11 +32,42 @@ public struct InvocationMessage: Sendable, Codable, CustomStringConvertible {
 }
 
 enum InvocationBehavior {
-    static func behavior(instance: some DistributedActor) -> _Behavior<InvocationMessage> {
+    static func behavior(instance weakInstance: Weak<some DistributedActor>) -> _Behavior<InvocationMessage> {
```

Comment:
part 2 of the fix, the behavior must not retain the actor; since the actor will then just always be retained as behaviors stay alive forever unless stopped.